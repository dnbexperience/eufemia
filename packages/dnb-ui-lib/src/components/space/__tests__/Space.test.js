/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../Space'

const snapshotProps = fakeProps(require.resolve('../Space'), {
  optional: true
})
snapshotProps.id = 'space'
snapshotProps.element = 'div'
snapshotProps.no_collapse = false

describe('Space component', () => {
  it('have to match snapshot', () => {
    const Comp = mount(<Component {...snapshotProps} />)
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('should have correct CSS classes', () => {
    const Comp = mount(<Component element="span" top="large" />)
    expect(
      Comp.find('span.dnb-space').hasClass('dnb-space__top--large')
    ).toBe(true)
  })

  it('should have collapse CSS classe', () => {
    const Comp = mount(<Component top="large" no_collapse={true} />)
    expect(Comp.find('.dnb-space--no-collapse').exists()).toBe(true)
  })
})

describe('Space scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-space.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-space-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
