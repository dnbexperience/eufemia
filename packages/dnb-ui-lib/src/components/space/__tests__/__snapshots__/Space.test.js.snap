// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space component have to match snapshot 1`] = `
<Space
  bottom="bottom"
  class="class"
  className="className"
  collapse={false}
  element="div"
  id="space"
  inline="inline"
  left="left"
  render_content={[Function]}
  right="right"
  top="top"
>
  <Element
    className="dnb-space className class"
    collapse={false}
    element="div"
  >
    <div
      className="dnb-space--no-collapse"
    >
      <div
        className="dnb-space className class"
      >
        children
      </div>
    </div>
  </Element>
</Space>
`;

exports[`Space scss have to match default theme snapshot 1`] = `
"/*
 * Np theme is provided
 *
 */
"
`;

exports[`Space scss have to match snapshot 1`] = `
"/*
* DNB Space
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-space {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: var(--font-size-basis);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-space *,
  .dnb-space ::before,
  .dnb-space ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-space ::before,
  .dnb-space ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
 * Space component
 *
 * TODO: Set spacing in the theme file, as theme comes later in the stack
 *
 */
.dnb-core-style .dnb-space__top--zero {
  margin-top: 0; }

.dnb-core-style .dnb-space__top--xx-small {
  margin-top: 0.25rem; }

.dnb-core-style .dnb-space__top--x-small {
  margin-top: 0.5rem; }

.dnb-core-style .dnb-space__top--small {
  margin-top: 1rem; }

.dnb-core-style .dnb-space__top--medium {
  margin-top: 1.5rem; }

.dnb-core-style .dnb-space__top--large {
  margin-top: 2rem; }

.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: 2.5rem; }

.dnb-core-style .dnb-space__top--x-large {
  margin-top: 3rem; }

.dnb-core-style .dnb-space__top--xx-large {
  margin-top: 3.5rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--x-small {
  margin-top: 4rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--small {
  margin-top: 4.5rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--medium {
  margin-top: 5rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--large {
  margin-top: 5.5rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: 6rem; }

.dnb-core-style .dnb-space__top--xx-large.dnb-core-style .dnb-space__top--x-large {
  margin-top: 6.5rem; }

.dnb-core-style .dnb-space__top--xx-large-x2 {
  margin-top: 7rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--x-small {
  margin-top: 7.5rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--small {
  margin-top: 8rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--medium {
  margin-top: 8.5rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--large {
  margin-top: 9rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--large.dnb-core-style .dnb-space__top--x-small {
  margin-top: 9.5rem; }

.dnb-core-style .dnb-space__top--xx-large-x2.dnb-core-style .dnb-space__top--x-large {
  margin-top: 10rem; }

.dnb-core-style .dnb-space__right--zero {
  margin-right: 0; }

.dnb-core-style .dnb-space__right--xx-small {
  margin-right: 0.25rem; }

.dnb-core-style .dnb-space__right--x-small {
  margin-right: 0.5rem; }

.dnb-core-style .dnb-space__right--small {
  margin-right: 1rem; }

.dnb-core-style .dnb-space__right--medium {
  margin-right: 1.5rem; }

.dnb-core-style .dnb-space__right--large {
  margin-right: 2rem; }

.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: 2.5rem; }

.dnb-core-style .dnb-space__right--x-large {
  margin-right: 3rem; }

.dnb-core-style .dnb-space__right--xx-large {
  margin-right: 3.5rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--x-small {
  margin-right: 4rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--small {
  margin-right: 4.5rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--medium {
  margin-right: 5rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--large {
  margin-right: 5.5rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: 6rem; }

.dnb-core-style .dnb-space__right--xx-large.dnb-core-style .dnb-space__right--x-large {
  margin-right: 6.5rem; }

.dnb-core-style .dnb-space__right--xx-large-x2 {
  margin-right: 7rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--x-small {
  margin-right: 7.5rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--small {
  margin-right: 8rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--medium {
  margin-right: 8.5rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--large {
  margin-right: 9rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--large.dnb-core-style .dnb-space__right--x-small {
  margin-right: 9.5rem; }

.dnb-core-style .dnb-space__right--xx-large-x2.dnb-core-style .dnb-space__right--x-large {
  margin-right: 10rem; }

.dnb-core-style .dnb-space__bottom--zero {
  margin-bottom: 0; }

.dnb-core-style .dnb-space__bottom--xx-small {
  margin-bottom: 0.25rem; }

.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 0.5rem; }

.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: 1rem; }

.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: 1.5rem; }

.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: 2rem; }

.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 2.5rem; }

.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: 3rem; }

.dnb-core-style .dnb-space__bottom--xx-large {
  margin-bottom: 3.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 4rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: 4.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: 5rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: 5.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 6rem; }

.dnb-core-style .dnb-space__bottom--xx-large.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: 6.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2 {
  margin-bottom: 7rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 7.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--small {
  margin-bottom: 8rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--medium {
  margin-bottom: 8.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--large {
  margin-bottom: 9rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--large.dnb-core-style .dnb-space__bottom--x-small {
  margin-bottom: 9.5rem; }

.dnb-core-style .dnb-space__bottom--xx-large-x2.dnb-core-style .dnb-space__bottom--x-large {
  margin-bottom: 10rem; }

.dnb-core-style .dnb-space__left--zero {
  margin-left: 0; }

.dnb-core-style .dnb-space__left--xx-small {
  margin-left: 0.25rem; }

.dnb-core-style .dnb-space__left--x-small {
  margin-left: 0.5rem; }

.dnb-core-style .dnb-space__left--small {
  margin-left: 1rem; }

.dnb-core-style .dnb-space__left--medium {
  margin-left: 1.5rem; }

.dnb-core-style .dnb-space__left--large {
  margin-left: 2rem; }

.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: 2.5rem; }

.dnb-core-style .dnb-space__left--x-large {
  margin-left: 3rem; }

.dnb-core-style .dnb-space__left--xx-large {
  margin-left: 3.5rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--x-small {
  margin-left: 4rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--small {
  margin-left: 4.5rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--medium {
  margin-left: 5rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--large {
  margin-left: 5.5rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: 6rem; }

.dnb-core-style .dnb-space__left--xx-large.dnb-core-style .dnb-space__left--x-large {
  margin-left: 6.5rem; }

.dnb-core-style .dnb-space__left--xx-large-x2 {
  margin-left: 7rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--x-small {
  margin-left: 7.5rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--small {
  margin-left: 8rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--medium {
  margin-left: 8.5rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--large {
  margin-left: 9rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--large.dnb-core-style .dnb-space__left--x-small {
  margin-left: 9.5rem; }

.dnb-core-style .dnb-space__left--xx-large-x2.dnb-core-style .dnb-space__left--x-large {
  margin-left: 10rem; }

/* stylelint-disable */
.dnb-space--inline {
  display: inline-block; }

.dnb-space--no-collapse {
  display: flow-root; }
  @media screen and (-ms-high-contrast: none) {
    .dnb-space--no-collapse {
      overflow: auto; } }

.dnb-space__top--zero {
  margin-top: 0; }

.dnb-space__top--xx-small {
  margin-top: 0.25rem; }

.dnb-space__top--x-small {
  margin-top: 0.5rem; }

.dnb-space__top--small {
  margin-top: 1rem; }

.dnb-space__top--medium {
  margin-top: 1.5rem; }

.dnb-space__top--large {
  margin-top: 2rem; }

.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: 2.5rem; }

.dnb-space__top--x-large {
  margin-top: 3rem; }

.dnb-space__top--xx-large {
  margin-top: 3.5rem; }

.dnb-space__top--xx-large.dnb-space__top--x-small {
  margin-top: 4rem; }

.dnb-space__top--xx-large.dnb-space__top--small {
  margin-top: 4.5rem; }

.dnb-space__top--xx-large.dnb-space__top--medium {
  margin-top: 5rem; }

.dnb-space__top--xx-large.dnb-space__top--large {
  margin-top: 5.5rem; }

.dnb-space__top--xx-large.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: 6rem; }

.dnb-space__top--xx-large.dnb-space__top--x-large {
  margin-top: 6.5rem; }

.dnb-space__top--xx-large-x2 {
  margin-top: 7rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--x-small {
  margin-top: 7.5rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--small {
  margin-top: 8rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--medium {
  margin-top: 8.5rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--large {
  margin-top: 9rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--large.dnb-space__top--x-small {
  margin-top: 9.5rem; }

.dnb-space__top--xx-large-x2.dnb-space__top--x-large {
  margin-top: 10rem; }

.dnb-space__right--zero {
  margin-right: 0; }

.dnb-space__right--xx-small {
  margin-right: 0.25rem; }

.dnb-space__right--x-small {
  margin-right: 0.5rem; }

.dnb-space__right--small {
  margin-right: 1rem; }

.dnb-space__right--medium {
  margin-right: 1.5rem; }

.dnb-space__right--large {
  margin-right: 2rem; }

.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: 2.5rem; }

.dnb-space__right--x-large {
  margin-right: 3rem; }

.dnb-space__right--xx-large {
  margin-right: 3.5rem; }

.dnb-space__right--xx-large.dnb-space__right--x-small {
  margin-right: 4rem; }

.dnb-space__right--xx-large.dnb-space__right--small {
  margin-right: 4.5rem; }

.dnb-space__right--xx-large.dnb-space__right--medium {
  margin-right: 5rem; }

.dnb-space__right--xx-large.dnb-space__right--large {
  margin-right: 5.5rem; }

.dnb-space__right--xx-large.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: 6rem; }

.dnb-space__right--xx-large.dnb-space__right--x-large {
  margin-right: 6.5rem; }

.dnb-space__right--xx-large-x2 {
  margin-right: 7rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--x-small {
  margin-right: 7.5rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--small {
  margin-right: 8rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--medium {
  margin-right: 8.5rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--large {
  margin-right: 9rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--large.dnb-space__right--x-small {
  margin-right: 9.5rem; }

.dnb-space__right--xx-large-x2.dnb-space__right--x-large {
  margin-right: 10rem; }

.dnb-space__bottom--zero {
  margin-bottom: 0; }

.dnb-space__bottom--xx-small {
  margin-bottom: 0.25rem; }

.dnb-space__bottom--x-small {
  margin-bottom: 0.5rem; }

.dnb-space__bottom--small {
  margin-bottom: 1rem; }

.dnb-space__bottom--medium {
  margin-bottom: 1.5rem; }

.dnb-space__bottom--large {
  margin-bottom: 2rem; }

.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: 2.5rem; }

.dnb-space__bottom--x-large {
  margin-bottom: 3rem; }

.dnb-space__bottom--xx-large {
  margin-bottom: 3.5rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--x-small {
  margin-bottom: 4rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--small {
  margin-bottom: 4.5rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--medium {
  margin-bottom: 5rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--large {
  margin-bottom: 5.5rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: 6rem; }

.dnb-space__bottom--xx-large.dnb-space__bottom--x-large {
  margin-bottom: 6.5rem; }

.dnb-space__bottom--xx-large-x2 {
  margin-bottom: 7rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--x-small {
  margin-bottom: 7.5rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--small {
  margin-bottom: 8rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--medium {
  margin-bottom: 8.5rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--large {
  margin-bottom: 9rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--large.dnb-space__bottom--x-small {
  margin-bottom: 9.5rem; }

.dnb-space__bottom--xx-large-x2.dnb-space__bottom--x-large {
  margin-bottom: 10rem; }

.dnb-space__left--zero {
  margin-left: 0; }

.dnb-space__left--xx-small {
  margin-left: 0.25rem; }

.dnb-space__left--x-small {
  margin-left: 0.5rem; }

.dnb-space__left--small {
  margin-left: 1rem; }

.dnb-space__left--medium {
  margin-left: 1.5rem; }

.dnb-space__left--large {
  margin-left: 2rem; }

.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: 2.5rem; }

.dnb-space__left--x-large {
  margin-left: 3rem; }

.dnb-space__left--xx-large {
  margin-left: 3.5rem; }

.dnb-space__left--xx-large.dnb-space__left--x-small {
  margin-left: 4rem; }

.dnb-space__left--xx-large.dnb-space__left--small {
  margin-left: 4.5rem; }

.dnb-space__left--xx-large.dnb-space__left--medium {
  margin-left: 5rem; }

.dnb-space__left--xx-large.dnb-space__left--large {
  margin-left: 5.5rem; }

.dnb-space__left--xx-large.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: 6rem; }

.dnb-space__left--xx-large.dnb-space__left--x-large {
  margin-left: 6.5rem; }

.dnb-space__left--xx-large-x2 {
  margin-left: 7rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--x-small {
  margin-left: 7.5rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--small {
  margin-left: 8rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--medium {
  margin-left: 8.5rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--large {
  margin-left: 9rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--large.dnb-space__left--x-small {
  margin-left: 9.5rem; }

.dnb-space__left--xx-large-x2.dnb-space__left--x-large {
  margin-left: 10rem; }

/* stylelint-enable */
"
`;
